runtime/primitives/src/currency.rs

```diff
@@ -10,6 +10,15 @@ use composable_support::rpc_helpers::FromHexStr;
 use serde::{Deserialize, Serialize};
 use sp_runtime::sp_std::ops::Deref;
 
+/// Trait used to write generalized code over well know currencies
+/// We use const to allow for match on these
+pub trait WellKnownCurrency<AssetId: Copy> {
```

Comment:
Do we have a reason to be generic here? Perhaps `pub trait WellKnownCurrency { const NATIVE: Self... }` is enough?

Reply:
i will check it.  so I doubt that currency will be also provider of well know CONST (consts kind of needed) in all cases. example, DOT vs KSM providers.  so will look - may be impl WellKnownCurrency<Self> will work.